core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -76,9 +76,8 @@ public ListenableFuture<Void> execute(
         }
 
         QualifiedObjectName targetTable = metadata.getRedirectedTableName(session, originalTableName);
-        boolean tableExists = metadata.listTables(session, targetTable.asQualifiedTablePrefix()).contains(targetTable);
 
-        if (!tableExists) {
+        if (!metadata.tableExists(session, targetTable)) {
```

Comment:
while reviewing `DropTableTask` code i'm thinking whether `isTable` would be a better name - in line with `isView` & `isMaterializedView` methods which already exist in `Metadata` . WDYT?

Reply:
As discussed in the standup, we will be relying on the `accessControl` check which will throw `TableNotFoundException` if the table does not exist.

Reply:
relying on `accessControl` did not help here.